composeApp/src/desktopMain/kotlin/edu/dyds/movies/di/MoviesDependencyInjector.kt

```diff
@@ -0,0 +1,67 @@
+package edu.dyds.movies.di
+
+import androidx.compose.runtime.Composable
+import androidx.lifecycle.viewmodel.compose.viewModel
+import edu.dyds.movies.data.MoviesRepositoryImpl
+import edu.dyds.movies.data.external.RemoteMoviesImpl
+import edu.dyds.movies.data.local.CacheMoviesImpl
+import edu.dyds.movies.domain.repository.MoviesRepository
+import edu.dyds.movies.domain.usecase.GetMovieDetailsUseCase
+import edu.dyds.movies.domain.usecase.GetMovieDetailsUseCaseImpl
+import edu.dyds.movies.domain.usecase.GetPopularMoviesUseCase
+import edu.dyds.movies.domain.usecase.GetPopularMoviesUseCaseImpl
+import edu.dyds.movies.presentation.detail.DetailScreenViewModel
+import edu.dyds.movies.presentation.home.HomeScreenViewModel
+import io.ktor.client.*
+import io.ktor.client.plugins.*
+import io.ktor.client.plugins.contentnegotiation.*
+import io.ktor.http.*
+import io.ktor.serialization.kotlinx.json.*
+import kotlinx.serialization.json.Json
+
+private const val API_KEY = "d18da1b5da16397619c688b0263cd281"
+
+object MoviesDependencyInjector {
+
+    private val tmdbHttpClient =
+        HttpClient {
+            install(ContentNegotiation) {
+                json(Json {
+                    ignoreUnknownKeys = true
+                })
+            }
+            install(DefaultRequest) {
+                url {
+                    protocol = URLProtocol.HTTPS
+                    host = "api.themoviedb.org"
+                    parameters.append("api_key", API_KEY)
+                }
+            }
+            install(HttpTimeout) {
+                requestTimeoutMillis = 5000
+            }
+        }
+
+    private var repository: MoviesRepository? = null
+
+    @Composable
+    fun getHomeScreenViewModel(): HomeScreenViewModel{
+        val getPopularMoviesUseCase:GetPopularMoviesUseCase = GetPopularMoviesUseCaseImpl(getRepository())
+        return viewModel { HomeScreenViewModel(getPopularMoviesUseCase)}
+    }
+
+    @Composable
+    fun getDetailScreenViewModel(): DetailScreenViewModel {
+        val getMovieDetailsUseCase:GetMovieDetailsUseCase = GetMovieDetailsUseCaseImpl(getRepository())
+        return viewModel { DetailScreenViewModel( getMovieDetailsUseCase)}
+    }
+
+    private fun getRepository(): MoviesRepository{
+        val repo: MoviesRepository = repository ?: MoviesRepositoryImpl(
+            CacheMoviesImpl(),
+            RemoteMoviesImpl(tmdbHttpClient)
+        )
+        repository= repo
+        return repo
+    }
```

Comment:
Se puede simplificar, no hace falta una función

```suggestion
    private var repository: MoviesRepository = MoviesRepositoryImpl(
            CacheMoviesImpl(),
            RemoteMoviesImpl(tmdbHttpClient)
        )
```